### ISSUE TITLE
`-C` is hard to pass to git commands

### ISSUE THREAD

**Issue by peff** [jonas/tig#910]
Since ed36f75c5b402114cb4937ea62b28b067723e672, trying to pass `-C` to `git blame` will get eaten by tig's `-C` parser. E.g.:

```
$ tig blame -C src/tig.c
tig: Failed to change directory to 
```

which used to invoke `git blame -C src/tig.c` prior to that commit. You can work around this with a `--`, but it took me some head-scratching to figure out where to put it:

```
$ tig blame -- -C src/tig.c
tig: Invalid number of options to blame
[...]

$ tig -- blame -C src/tig.c
[works!]
```

Should we perhaps only respect `-C` before hitting a sub-command name?

/cc @koutcher

**Comment by peff** [comment]
> `$ tig -- blame -C src/tig.c`

Actually, never mind. This does not work at all. :wink: I just saw that it did not barf and declared victory, but of course it means something entirely different (it is a log on the pathspecs `blame`, `-C`, etc). So I don't think there is any way to pass `-C` at all now.

**Comment by koutcher** [comment]
The problem is the space between -C and the path. Tig’s command line parser is very primitive and doesn’t allow it. The same applies to tig -S. It is certainly something to improve in the future.

**Comment by peff** [comment]
For actually trying to use `tig -C`, the space may be a problem. But in this case I do not want tig to parse `-C` at all. I want it to pass it to `git blame`, as it used to (where `-C` does not take an argument at all).

**Comment by peff** [comment]
So I think something like https://github.com/jonas/tig/pull/911 is the right solution. I don't know quite how you were using `-C`. It does change things if you were used to using `-C` after the sub-command (you'll have to put it before the subcommand now), but that's unavoidable. We need some way to distinguish which options are meant for tig and which for git.

**Comment by koutcher** [comment]
The subcommand is only checked in argv[1]. How about `tig blame -- -C` ?

**Comment by koutcher** [comment]
#911 makes sense yes.

**Comment by peff** [comment]
Fixed by #911.
